concordia/views.py

```diff
@@ -473,12 +474,128 @@ def annotate_children_with_progress_stats(children):
         obj.lowest_transcription_status = lowest_status
 
 
+@method_decorator(default_cache_control, name="dispatch")
+class TopicListView(APIListView):
+    template_name = "transcriptions/topic_list.html"
+    paginate_by = 10
+    queryset = Topic.objects.published().order_by("ordering", "title")
+    context_object_name = "topics"
+
+    def serialize_context(self, context):
+        data = super().serialize_context(context)
+
+        object_list = data["objects"]
+
+        status_count_keys = {
+            status: f"{status}_count" for status in TranscriptionStatus.CHOICE_MAP
+        }
+
+        topic_stats_qs = (
+            Topic.objects.filter(pk__in=[i["id"] for i in object_list])
+            .annotate(
+                **{
+                    v: Count(
+                        "project__item__asset",
+                        filter=Q(
+                            project__published=True,
+                            project__item__published=True,
+                            project__item__asset__published=True,
+                            project__item__asset__transcription_status=k,
+                        ),
+                    )
+                    for k, v in status_count_keys.items()
+                }
+            )
+            .values("pk", *status_count_keys.values())
+        )
+
+        topic_asset_counts = {}
+        for topic_stats in topic_stats_qs:
+            topic_asset_counts[topic_stats.pop("pk")] = topic_stats
+
+        for obj in object_list:
+            obj["asset_stats"] = topic_asset_counts[obj["id"]]
+
+        return data
+
+
+@method_decorator(default_cache_control, name="dispatch")
+class CampaignTopicListView(TemplateView):
+    template_name = "transcriptions/campaign_topic_list.html"
+
+    def get(self, context):
+        data = {}
+        data["campaigns"] = Campaign.objects.published().order_by("ordering", "title")
+        data["topics"] = Topic.objects.published().order_by("ordering", "title")
+
+        return render(self.request, self.template_name, data)
+
+
+@method_decorator(default_cache_control, name="dispatch")
+class TopicDetailView(APIDetailView):
+    template_name = "transcriptions/topic_detail.html"
+    context_object_name = "topic"
+    queryset = Topic.objects.order_by("title")
```

Comment:
Do we want to make this use `.published()` so it'll 404 if you ask it for an unpublished topic?

Reply:
Yes you're right, I'll fix that.